### ISSUE TITLE
Like / add to music actions (feature request)

### ISSUE THREAD

**Issue by alexanderjeurissen** [caleb531/play-song#37]
I really love your workflow ! great work.

I find myself however missing some features, although I'm aware that this workflow is geared towards playing and queuing songs it would be great if the following use-cases could be implemented in this workflow. Note that some of those probably only apply to Apple music subscribers.

1. Liking / disliking the current playing song 
2. Play more like this / Play less like this (great when you are listening to a artist/song radio)
3. Add to My Music

I'm no apple script expert, But I'm certainly willing to help out with this. Let me know what you think of the proposed features.

**Comment by caleb531** [comment]
As far as I know and from what I've tested, AppleScript doesn't support recently-added iTunes (like Play Next). Therefore, because the workflow is at the mercy of AppleScript's capabilities, Play Song is unable to adopt any of the features you mentioned. Sorry :-/

Caleb

**Comment by alexanderjeurissen** [comment]
I've looked into it some more and the issue is that non-local files like apple music can't be accessed through applescript / automator workflows.

See http://www.openradar.me/21674162 describing the bug and the following article from Doug http://dougscripts.com/itunes/2015/07/itunes-12-2-and-applescript-issues-volume-2/

I also attached a screenshot showing the issue, the object you receive when getting current track info is null and thus can't be referenced / used.

![screen shot 2016-02-07 at 18 17 20](https://cloud.githubusercontent.com/assets/1220084/12873907/c4e81050-cdc9-11e5-927e-ee3761e6ecaf.png)
![screen shot 2016-02-07 at 18 17 31](https://cloud.githubusercontent.com/assets/1220084/12873906/c2254fb8-cdc9-11e5-9a00-7f66fd03d0cf.png)

The functionality works with locally stored music though, the "Add to playlist" workflow of Automator supports adding to the "Library" which I assume is the equivalent of MyMusic.

If apple fixes the issues above, are you willing to implement the features I suggested ?
